### PR TITLE
Adding tree selector for selecting automate methods

### PR DESCRIPTION
**Pull request by romanblanco**
## Needs to be merged together with https://github.com/ManageIQ/manageiq-ui-classic/pull/2591

Replacing temporery input box with treeSelector component for selecting automate methods

Links
----------------
* https://bugzilla.redhat.com/show_bug.cgi?id=1507066
* https://www.pivotaltracker.com/story/show/147779457

Steps for Testing/QA
-------------------------------
screencast from 2017-11-02 13-00-11

@skateman @Hyperkid123 @karelhala can you check it?

### CODE REVIEW

File: src/dialog-editor/components/modal-field-template/modalFieldTemplateComponent.ts [hunk 1]
```diff
@@ -48,5 +48,11 @@ export default class ModalFieldTemplate {
     resolveCategories: '=?',
     modalTabIsSet: '<',
     modalTab: '=',
+    lazyLoad: '<',
+    onSelect: '<',
+    treeSelectorData: '=',
+    treeSelectorToggle: '<',
+    treeSelectorShow: '=',
+    treeSelectorIncludeDomain: '=',
```
Comment (skateman) [inline src/dialog-editor/components/modal-field-template/modalFieldTemplateComponent.ts]: Do you really need the 2way binding?

File: src/dialog-editor/components/modal/modalComponent.ts [hunk 2]
```diff
@@ -310,6 +316,12 @@ export default class Modal {
   public transclude = true;
   public controller: any = ModalController;
   public bindings: any = {
+    lazyLoad: '<',
+    onSelect: '<',
+    treeSelectorData: '=',
+    treeSelectorToggle: '<',
+    treeSelectorShow: '=',
+    treeSelectorIncludeDomain: '=',
```
Comment (skateman) [inline src/dialog-editor/components/modal/modalComponent.ts]: same as above

File: src/dialog-editor/components/modal-field/field.html [hunk 3]
```diff
@@ -90,6 +126,12 @@ <h4 class="modal-title" id="myModalLabel" translate>Edit Field Details</h4>
     </dialog-editor-modal-field-template>
     <dialog-editor-modal-field-template ng-switch-when="DialogFieldDropDownList"
                                         template="drop-down-list.html"
+                                        lazy-load="vm.lazyLoad"
+                                        tree-selector-data="vm.treeSelectorData"
+                                        tree-selector-toggle="vm.treeSelectorToggle"
+                                        tree-selector-show="vm.treeSelectorShow"
+                                        tree-selector-include-domain="vm.treeSelectorIncludeDomain"
+                                        on-select="vm.onSelect"
```
Comment (karelhala) [inline src/dialog-editor/components/modal-field/field.html]: I really don't like how this is repeated all the times, but it's not part of this PR.

**Comment by Hyperkid123** [comment]
Seems to be working fine. Definitely agree with @karelhala about, the duplicate code. That should be something changed somewhere in the future.

**Comment by martinpovolny** [comment]
Needs to go to Gaprindashivi.

**Comment by himdel** [comment]
Released in manageiq-ui-components 1.0.1.

https://github.com/ManageIQ/manageiq-ui-classic/pull/2839
